pkg/maps/lbmap/lbmap.go

```diff
@@ -406,22 +406,20 @@ func (*LBBPFMap) UpdateSourceRanges(revNATID uint16, prevSourceRanges []*cidr.CI
 
 	srcRangeMap := map[string]*cidr.CIDR{}
 	for _, cidr := range sourceRanges {
-		// k8s api server does not catch the IP family mismatch, so we need to catch it here
+		// k8s api server does not catch the IP family mismatch, so we need
+		// to catch it here and below.
 		if ip.IsIPv6(cidr.IP) == !ipv6 {
-			log.WithFields(logrus.Fields{
-				logfields.ServiceID: revNATID,
-				logfields.CIDR:      cidr,
-			}).Warn("Source range's IP family does not match with the LB's. Ignoring the source range CIDR")
 			continue
 		}
 		srcRangeMap[cidr.String()] = cidr
 	}
 
 	for _, prevCIDR := range prevSourceRanges {
+		if ip.IsIPv6(prevCIDR.IP) == !ipv6 {
+			continue
+		}
 		if _, found := srcRangeMap[prevCIDR.String()]; !found {
-			if err := m.Delete(srcRangeKey(prevCIDR, revNATID, ipv6)); err != nil {
-				return err
-			}
+			m.Delete(srcRangeKey(prevCIDR, revNATID, ipv6))
```

Comment:
Can we log a message when the deletion fails and let users know that stale entries remain?

Reply:
it would fail if the key was not found, so it is not clear if there are stale entries or not. I'd rather not confuse the user.